### PR TITLE
Use a variable to enable aws:ec2:vpc:AssociatePublicIpAddress

### PR DESCRIPTION
**Pull request by guikcd**
## what
* This pull request just adds a variable to enable the automatic creation of Elastic IP attached to instances.

## why
* If there is not NAT gateway in `private_subnets` (but an Internet gateway) there is a need to attach Elastic IP to instances in order to communicate with Elastic BeanStalk publics endpoints.
* I agree with the default value set before, `false`; a correct setup has a *NAT gateway* in a *private* subnet :)

## sources
* https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/vpc.html
* https://docs.aws.amazon.com/elasticbeanstalk/latest/dg/vpc-no-nat.html

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -352,7 +352,7 @@ resource "aws_elastic_beanstalk_environment" "default" {
   setting {
     namespace = "aws:ec2:vpc"
     name      = "AssociatePublicIpAddress"
-    value     = "false"
+    value     = "${var.association_public_ipaddress}"
```
Comment (osterman) [inline main.tf]: Please update to `association_public_ip_address` so that it corresponds to the camel case notation.

**Comment by guikcd** [comment]
`s/association_public_ipaddress/association_public_ip_address/` done!

File: variables.tf [hunk 2]
```diff
@@ -127,6 +127,11 @@ variable "instance_type" {
   description = "Instances type"
 }
 
+variable "association_public_ip_address" {
```
Comment (aknysh) [inline variables.tf]: @guikcd  looks good, but why don't we call it `associate_public_ip_address`?

Comment (osterman) [reply variables.tf]: crap, that was my fault! I typo'd it.

**Comment by guikcd** [comment]
`s/association_public_ip_address/associate_public_ip_address/` done!

**Pull request merged**